api/v1/saml2aws_types.go

```diff
@@ -57,6 +57,7 @@ type Saml2AwsSpec struct {
 type Saml2AwsStatus struct {
 	RoleARN        string      `json:"roleARN,omitempty"`
 	ExpirationTime metav1.Time `json:"expirationTime,omitempty"`
+	State          string      `json:"status,omitempty"`
```

Comment:
IIUC from https://github.com/AbsaOSS/samlet/issues/32, the request was to add the State (`Provisioning/Error/Completed`), and the associated message in case of Error.
So I guess there should be two fields, `State` and `Error`

Reply:
IMO it makes no sense to loose reconcile cycle and spam API with `provisioning` message as state changing almost immediately. Unless watched a user have no chance to see initial state. This leads into actual two states:
* current step it failed on including error message
* or `Completed`
Having separate field for error message is an overkill